tests/ctst/steps/quotas/quotas.ts

```diff
@@ -0,0 +1,150 @@
+/* eslint-disable no-case-declarations */
+import fs from 'fs';
+import lockFile from 'proper-lockfile';
+import { createHash } from 'crypto';
+import { Given, Before, When } from '@cucumber/cucumber';
+import Zenko from '../../world/Zenko';
+import { Scality, Command, CacheHelper, Constants, Utils } from 'cli-testing';
+import { createJobAndWaitForCompletion } from 'steps/utils/kubernetes';
+import { createBucketWithConfiguration, putObject } from 'steps/utils/utils';
+
+function hashStringAndKeepFirst20Characters(input: string) {
+    return createHash('sha256').update(input).digest('hex').slice(0, 20);
+}
+
+/**
+ * The objective of this hook is to prepare all the buckets and accounts
+ * we use during quota checks, so that we avoid running the job multiple
+ * times, which affects the performance of the tests.
+ * The steps arer: create an account, then create a simple bucket
+ */
+Before({tags: '@Quotas'}, async function ({ gherkinDocument, pickle }) {
+    let initiated = false;
+    let releaseLock: (() => Promise<void>) | false = false;
+    const output: { [key: string]: { AccessKey: string, SecretKey: string }} = {};
+    const world = this as Zenko;
+
+    await Zenko.init(world.parameters);
+
+    const featureName = gherkinDocument.feature?.name?.replace(/ /g, '-').toLowerCase() || 'quotas';
+    const filePath = `/tmp/${featureName}`;
+
+    if (!fs.existsSync(filePath)) {
+        fs.writeFileSync(filePath, JSON.stringify({
+            ready: false,
+        }));
+    } else {
+        initiated = true;
+    }
+
+    if (!initiated) {
+        try {
+            releaseLock = await lockFile.lock(filePath, { stale: Constants.DEFAULT_TIMEOUT / 2 });
+        } catch (err) {
+            releaseLock = false;
+        }
+    }
+
+    if (releaseLock) {
+        const isBucketNonVersioned = gherkinDocument.feature?.tags?.find(
+            tag => tag.name === 'NonVersioned') === undefined;
+        for (const scenario of gherkinDocument.feature?.children || []) {
+            for (const example of scenario.scenario?.examples || []) {
+                for (const values of example.tableBody || []) {
+                    const scenarioWithExampleID = hashStringAndKeepFirst20Characters(`${values.id}`);
+                    await world.createAccount(scenarioWithExampleID, true);
+                    await createBucketWithConfiguration(world, scenarioWithExampleID,
+                        isBucketNonVersioned ? '' : 'with');
+                    await putObject(world);
+                    output[scenarioWithExampleID] = {
+                        AccessKey: CacheHelper.parameters?.AccessKey || Constants.DEFAULT_ACCESS_KEY,
+                        SecretKey: CacheHelper.parameters?.SecretKey || Constants.DEFAULT_SECRET_KEY,
+                    };
+                }
+            }
+        }
+    
+        await createJobAndWaitForCompletion(world, 'end2end-ops-count-items', 'quotas-setup');
+        await Utils.sleep(2000);
```

Comment:
Why are we waiting 2 seconds here ? Might be worth a comment to explain if really needed, and is it parameterizable / calculable instead of hardcoding it

Reply:
Added a comment